google/cloud/storage/internal/grpc_client.cc

```diff
@@ -206,7 +206,12 @@ int DefaultGrpcNumChannels(std::string const& endpoint) {
 
 Options DefaultOptionsGrpc(Options options) {
   using ::google::cloud::internal::GetEnv;
-
+  // Experiments show that gRPC gets better upload throughput when the upload
+  // buffer is larger than 32MiB.
+  auto constexpr kDefaultGrpcUploadBufferSize = 32 * 1024 * 1024L;
```

Comment:
Nit: I presume you meant "at least 32MiB", otherwise we're setting the default to something we know could be better?